public/components/DiscoverAction/SuggestAnomalyDetector.tsx

```diff
@@ -0,0 +1,859 @@
+        resultIndex: undefined,
+        filters: [],
+        featureList: [] as FeaturesFormikValues[],
+        categoryFieldEnabled: false,
+        categoryField: [] as string[],
+        realTime: true,
+        historical: false,
+    };
+
+    return (
+        <div className="add-anomaly-detector">
+            <Formik
+                initialValues={initialDetectorValue}
+                onSubmit={handleSubmit}
+                validateOnChange={true}
+                validate={validateFeatures}
+            >
+                {(formikProps) => (
+                    <>
+                        <EuiFlyoutHeader>
+                            <EuiTitle size='s'>
+                                <h2 id="add-anomaly-detector__title">
+                                    Suggest anomaly detector
+                                </h2>
+                            </EuiTitle>
+                            <EuiSpacer size="m" />
+                            <EuiText size='xs'>
+                                Create an anomaly detector based on the parameters(model features and categorical field) suggested by OpenSearch Assistant.
+                            </EuiText>
+                        </EuiFlyoutHeader>
+                        <EuiFlyoutBody>
+                            <div className="create-new">
+                                <EuiTitle size="s">
+                                    <h3>Detector details</h3>
+                                </EuiTitle>
+                                <EuiSpacer size="m" />
+
+                                <EnhancedAccordion
+                                    id="detectorDetailsAccordion"
+                                    title={detectorName}
+                                    isOpen={accordionsOpen.detectorDetails}
+                                    onToggle={() => onAccordionToggle('detectorDetails')}
+                                    subTitle={
+                                        <EuiText size="m">
+                                            <p>
+                                                Detector interval: {intervalValue} minute(s); Window
+                                                delay: {delayValue} minute(s)
+                                            </p>
+                                        </EuiText>
+                                    }
+                                >
+                                    <Field name="name" validate={validateAnomalyDetectorName}>
+                                        {({ field, form }: FieldProps) => (
+                                            <FormattedFormRow
+                                                title="Name"
+                                                isInvalid={isInvalid(field.name, form)}
+                                                error={getError(field.name, form)}
+                                            >
+                                                <EuiFieldText
```

Comment:
How are you populating the detector name? Based on the screenshot, it looks like the detector name doesn't follow any kind of UUID naming convention. We don't allow duplicated detector names, how is this being handled here? 

By the way, the demo recording video is broken in the PR description. Please consider reuploading it

Reply:
Thanks for pointing that, currently it will throw an error if the detector name is duplicated, but I think the behavior should be consistent with the feature anywhere, I've changed the code to reuse the existing generate detector name method.
I've uploaded the video again, please check that.